### ISSUE TITLE
Can't render Array/List as Root JSON object using StreamingJsonBuilder

### ISSUE THREAD

**Issue by valerius** [grails/grails-core#10050]
Thanks so much for the great work! Just writing to inform of an issue I ran into when upgrading from 2.4. Basically, The change to StreamingJsonBuilder prevents rendering of Array as root. That is:

```groovy
render(contentType: 'application/json') {
   [[title: 'foo'], [title: 'bar']] // this doesn't work anymore because of StreamingJsonBuilder introduction. The problem is, there is no practical alternative
}
```

```groovy
 render   ([[title: 'foo'], [title: 'bar']] as JSON) // this form works
```

Both of which  should return:

```json
[{"title":"foo"},{"title":"bar"}]
```

But does not because StreamingJsonBuilder always adds CurlyBraces, which prevents array as root object:

**StreamingJsonBuilder
```groovy
    public Object call(@DelegatesTo(StreamingJsonDelegate.class) Closure c) throws IOException {
        writer.write(JsonOutput.OPEN_BRACE); // prevents Array support
        StreamingJsonDelegate.cloneDelegateAndGetContent(writer, c);
        writer.write(JsonOutput.CLOSE_BRACE);  // prevents Array support
        return null;
    }
```

as it is instantiated like this:

```groovy
private void renderJsonInternal(HttpServletResponse response, Closure callable) {
    if( Holders.getConfig()?.getProperty(Settings.SETTING_LEGACY_JSON_BUILDER, Boolean.class, false) ) {
        def builder = new JSONBuilder()
        JSON json = builder.build(callable) 
        json.render response
    }
    else {
        response.setContentType(GrailsWebUtil.getContentType(MimeType.JSON.getName(), response.getCharacterEncoding() ?: "UTF-8"))
        def jsonBuilder = new StreamingJsonBuilder(response.writer) 
        jsonBuilder.call callable // call method above which hard codes braces
    }
}
```

If there is another way to have arrays as root, that would be great. Otherwise, it would seem like a bug?

**Comment by graemerocher** [comment]
You can using streaming json builder directly:

```groovy
            def jsonBuilder = new StreamingJsonBuilder(response.writer)
            jsonBuilder.call( [[title: 'foo'], [title: 'bar']])
```

**Comment by valerius** [comment]
Thanks so much for the quick reply! Correct, I have played with using it directly, but am baffled how I can do it cleanly within the context of render, which is why I think its a bug, not with StreamingJSonBuilder necessarily, as I got to work in the console in the  manner you provided, but with render, which now unlike before, seems kind of awkward in this context; and therefore a bug IMHO, especially in view of the legacy version which can still be enabled, but per documentation is not the recommended way (even though I prefer it). I choose the render closure because I may need to change the content type, response code, and other request headers, which 'render as JSON' doesn't appear to allow me to do. Wherefore, unless it can be done this way, though its possible to get around unnaturally, it doesn't read very well and therefore consider  it a bug (especially when it use to work)
```
 
render(contentType: 'text/json',status: HttpServletResponse.SC_CREATED) {
   // how would I do it here?
   // notice I changed the contentType AND status and it looks clean
}
```

Just my two cents anyways. Keep up the good work :)
